### ISSUE TITLE
i lost all my container when i restart my rancher 

### ISSUE THREAD

**Issue by Pwapou** [rancher/rancher#10397]
**Rancher versions:**
rancher/server: 1.6.10
rancher/agent: 1.2.6

**Infrastructure Stack versions:**
healthcheck: v0.3.3
ipsec: 
network-services: 0.9.4
scheduler: 0.8.2
kubernetes (if applicable):

**Docker version: (`docker version`,`docker info` preferred)** 
17.09.0-ce
**Operating system and kernel: (`cat /etc/os-release`, `uname -r` preferred)**
Ubuntu 16.04.3 LTS
**Type/provider of hosts: (VirtualBox/Bare-metal/AWS/GCE/DO)**
Bare-metal OVH
**Setup details: (single node rancher vs. HA rancher, internal DB vs. external DB)**
single node rancher 
**Environment Template: (Cattle/Kubernetes/Swarm/Mesos)**
Cattle
**Steps to Reproduce:**
**Results:**
Hi, 
When i restart my bare-metal serveur and restart my rancher server container, i lost my agent and all my stack... how to restart a single-node and never lost all services ?

**Comment by superseb** [comment]
All state is, by default, stored in the embedded MySQL database, which stores it's data on a VOLUME defined in the Dockerfile (https://github.com/rancher/rancher/blob/v1.6.10/server/Dockerfile#L73). This volume is stored in `/var/lib/docker` (by default). If you restart the server (`reboot`) or restart the container (`docker restart container_id_or_name`), they created volume is still associated with the container and will be used. If `/var/lib/docker` is cleared, or if you start a new container (`docker run`), a new volume will be created and there will be no state to be re-used.

If you want to be more flexible, you can bind-mount a host directory to the MySQL directory (http://rancher.com/docs/rancher/v1.6/en/installing-rancher/installing-server/#single-container-bind-mount) or use an external database (http://rancher.com/docs/rancher/v1.6/en/installing-rancher/installing-server/#single-container-external-database).

Closing as I don't think this is a bug, let me know if you have any other questions.

**Comment by jonathascarrijo** [comment]
Same thing happened to me while running Rancher on a RancherOS machine.
I was using binding, which can be verified in the output of docker inspect:
```
...
        "HostConfig": {
            "Binds": [
                "/var/lib/mysql/:/var/lib/mysql"
            ],
...
```
I was trying to solve issue #11142 , and someone suggested I restarted rancher.
I first did a docker restart to the rancher container, which didn't solve my problem (but didn't erase my configurations either).
Then I tried to reboot the server (RancherOS on DigitalOcean).
When I accessed Rancher again, it was a fresh install, and I realized I lost all my staging and production environments.
